Data-Structures/src/test/java/stacksandqueues/AnimalShelterTest.java

```diff
@@ -0,0 +1,93 @@
+package stacksandqueues;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class AnimalShelterTest {
+    @Test(expected = IllegalArgumentException.class)
+    public void testExceptionThrow(){
+        AnimalShelter test1 = new AnimalShelter();
+        test1.dequeuePet("Elephant");
+    }
+
+
+    @Test public void testEnqueue(){
+        AnimalShelter test2 = new AnimalShelter();
+        Dog mee = new Dog("bingo");
+
+        test2.enqueuePet(mee);
+
+        assertEquals("bingo", test2.dequeuePet("dog").name);
+    }
+    @Test public void testDequeue(){
+        AnimalShelter test2 = new AnimalShelter();
+        Dog mee = new Dog("bingo");
+        Dog me = new Dog("bin");
+        Dog m = new Dog("b");
```

Comment:
Next time try to come up with better names here, it was a bit hard to read and figure out what was being tested. `dog1`, `dog2`, `dog3` are fine, or do it the same as the dog's name.